includes/Core/Authentication/Clients/OAuth_Client.php

```diff
@@ -344,7 +344,15 @@ function ( $scope ) {
 			$additional_scopes = array();
 		}
 
-		$redirect_url = add_query_arg( array( 'notification' => 'authentication_success' ), $redirect_url );
+		$parts = URL::parse( $redirect_url );
```

Comment:
How about we pull out the `PHP_URL_QUERY` directly rather than all the parts as its the only part we need?

Reply:
Thanks for pointing that out - good shout. I've made the change :+1: